hashes_and_arrays.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+class Library
+  @books = []
+
+  book1 = {
+    id: @books.length + 1,
+    year: 1956,
+    status: 'available',
+    author: 'Lucy Maud Montgomery',
+    name: 'Anne of Green Gables'
+  }
+  @books << book1
+
+  book2 = {
+    id: @books.length + 1,
+    year: 1985,
+    status: 'unavailable',
+    author: 'Edith Eva Eger',
+    name: 'The Choice'
+  }
+  @books << book2
+
+  book3 = {
+    id: @books.length + 1,
+    year: 2020,
+    status: 'available',
+    author: 'Edith Eva Eger',
+    name: 'The Gift'
+  }
+  @books << book3
+
+  def find_book(search_variable)
+    @books.find { |book| book.values.include?(search_variable) }
```

Comment:
If I want to find all books by some author I will have only this author's first book. 
I would like that you return here an array of all books, not only the first one. 
````
def find_book(search_key, search_value)
# Try to search books not only by value, because if the author's name is the name of the book, this method could return wrong results.
end
```